src/cython/sktda/preprocessing.py

```diff
@@ -0,0 +1,274 @@
+                diag_pad[:num_pts, dim] = np.ones(num_pts)
+                Xfit.append(diag_pad)                    
+        else:
+            Xfit = X
+        return Xfit
+
+class ProminentPoints(BaseEstimator, TransformerMixin):
+    """
+    This is a class for removing points that are close or far from the diagonal in persistence diagrams.
+    """
+    def __init__(self, use=False, num_pts=10, threshold=-1, location="upper", point_type="finite"):
+        """
+        Constructor for the ProminentPoints class.
+     
+        Attributes:
+            use (bool): whether to use the class or not (default False).
+            location (string): either "upper" or "lower" (default "upper"). Whether to keep the points that are far away ("upper") or close ("lower") to the diagonal.
+            num_pts (int): cardinality threshold (default 10). If location == "upper", keep the top **num_pts** points that are the farthest away from the diagonal. If location == "lower", keep the top **num_pts** points that are the closest to the diagonal. 
+            threshold (double): distance-to-diagonal threshold (default -1). If location == "upper", keep the points that are at least at a distance **threshold** from the diagonal. If location == "lower", keep the points that are at most at a distance **threshold** from the diagonal. 
+            point_type (string): either "finite" if persistence diagrams are n x 2 numpy arrays, or "essential" if persistence diagrams are n x 1 numpy arrays (default "finite"). If "finite", points are ordered and thresholded by distance-to-diagonal. If "essential", points are ordered and thresholded by first coordinate.
+        """
+        self.num_pts    = num_pts
+        self.threshold  = threshold
+        self.use        = use
+        self.location   = location
+        self.point_type = point_type
+
+    def fit(self, X, y=None):
+        """
+        Fit the ProminentPoints class on a list of persistence diagrams (this function actually does nothing but is useful when ProminentPoints is included in a scikit-learn Pipeline).
+
+        Parameters:
+            X (list of n x 2 or n x 1 numpy arrays): input persistence diagrams.
+            y (n x 1 array): persistence diagram labels (unused).
+        """
+        return self
+
+    def transform(self, X):
+        """
+        If location == "upper", first select the top **num_pts** points that are the farthest away from the diagonal, then select and return from these points the ones that are at least at distance **threshold** from the diagonal for each persistence diagram individually. If location == "lower", first select the top **num_pts** points that are the closest to the diagonal, then select and return from these points the ones that are at most at distance **threshold** from the diagonal for each persistence diagram individually. If point_type == "essential", do the same with first coordinate instead of distance-to-diagonal.
```

Comment:
I don't see any sorting in the essential case.

Reply:
Yes you are right, sorting is not justified here because it is hard to rank the diagram points (as opposed to the finite case, where the distance to the diagonal has a meaning).

Reply:
So, should the doc say that for essential points (aka nx1 arrays) we return an arbitrary subset of num_pts points? Actually, no, the code seems to compare the birth time to the threshold?